src/IB/IIMethod.cpp

```diff
@@ -2026,15 +2081,15 @@ IIMethod::extrapolatePressureForTraction(const int p_data_idx, const double data
 
     if (MathUtilities<double>::equalEps(data_time, d_current_time))
     {
-        X_vec = d_X_current_vecs[part];
+        X_vec = getMeshCoordinates(d_use_current_mesh_configuration_for_interactions, "current", part);
     }
     else if (MathUtilities<double>::equalEps(data_time, d_half_time))
     {
-        X_vec = d_X_half_vecs[part];
+        X_vec = getMeshCoordinates(d_use_current_mesh_configuration_for_interactions, "half", part);
     }
     else if (MathUtilities<double>::equalEps(data_time, d_new_time))
     {
-        X_vec = d_X_new_vecs[part];
+        X_vec = getMeshCoordinates(d_use_current_mesh_configuration_for_interactions, "new", part);
     }
```

Comment:
Same comment as above --- can collapse down to a single call to `getMeshCoordinates` that passes in `data_time`.